src/cmd/remove.nim

```diff
@@ -5,7 +5,22 @@ import ../cli/prompt, ../cli/term
 import ../modpack/files, ../modpack/install
 import ../util/flow
 
-proc paxRemove*(name: string): void =
+proc removeDependencies(manifest: var Manifest, file: ManifestFile): void =
+  ## Recursively removes dependencies of a mod
+  if len(file.dependencies) > 0:
+    for id in file.dependencies:
+      if not manifest.isInstalled(id):
+        continue
+
+      let modToRemove = manifest.getFile(id)
+
+      if not modToRemove.explicit:
```

Comment:
You also need to check if the `modToRemove` is a dependency of other mods than `file`, because if it is, it should not be removed.